### ISSUE TITLE
"Program terminating" with oldFactArg and dot products

### ISSUE THREAD

**Issue by t-hahn** [vermaseren/form#97]
I'm getting program aborts ("Program terminating in line ...") if I use factarg with oldFactArg on expressions containing dot products.  See here:
`v e1, e2, k1, k2;`
`s a, b;`
`cf dotM;`
`L testbad = dotM(e1.k1*e2.k1);`
`L testok = dotM(a*b);`
`.sort`
`on oldFactArg;`
`factarg dotM;`
`print;`
`.end`

Best, Thomas

**Comment by t-hahn** [comment]
Forgot to mention: this happens with form/tform compiled from yesterday's sources.
I checked with the version of April 19 that I was using before and there it doesn't happen.

**Comment by tueda** [comment]
Here is another example of the crash of OldFactArg:
```
On OldFactArg;
V p1,p2,p3,p4;
S x;
CF f;
T t;
L OK1 = f(t(p1)*x);
L OK2 = f(t(p1,p2)*x);
L OK3 = f(t(p1,p2,p3)*x);
L BAD = f(t(p1,p2,p3,p4)*x);
factarg f;
P;
.end
```

**Comment by tueda** [comment]
Introduced by 29e608ee0e.

**Comment by tueda** [comment]
Somehow fixed by 10eada9.
